tools/stress-cluster/services/chaos-watcher/ChaosResources/GenericChaosClient.cs

```diff
@@ -0,0 +1,56 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using k8s;
+
+namespace chaos_watcher
+{
+    public class GenericChaosClient
+    {
+        // Plural names for listing resource instances via Kubernetes API
+        public enum ChaosResourcePlurals
+        {
+            networkchaos,
+            stresschaos,
+            httpchaos,
+            iochaos,
+            kernelchaos,
+            timechaos,
+            jvmchaos
+        };
+
+        public string Group = "chaos-mesh.org";
+        public string Version = "v1alpha1";
+
+        private List<GenericClient> Clients = new List<GenericClient>();
+
+        public GenericChaosClient(KubernetesClientConfiguration config)
+        {
+            foreach (var plural in Enum.GetValues(typeof(ChaosResourcePlurals)))
+            {
+                Clients.Add(new GenericClient(config, Group, Version, plural.ToString()));
+            }
+        }
+
+        public async Task<CustomResourceList<GenericChaosResource>> ListNamespacedAsync(string ns)
+        {
+            var tasks = Clients.Select(cl => 
+            {
+                return Task.Run(async() => await cl.ListNamespacedAsync<CustomResourceList<GenericChaosResource>>(ns));
```

Comment:
Is the extra Task.Run needed here? I assume cl.ListNamespacedAsync returns a Task already.

Reply:
I'm rusty on the `Task` internals + how they interact with linq. If I just do await, will it return a task but still block? I want to spawn all list calls concurrently and then wait on the results.

Reply:
Oh you mean `ListNamespacedAsync` will handle that under the hood since it's an async function. I'll change it.